File: src/main/java/com/kisman/cc/loader/security/UnsafeProvider.java
package com.kisman.cc.loader.security;

import sun.misc.Unsafe;

import java.lang.reflect.Constructor;
import java.lang.reflect.Field;

public interface UnsafeProvider {
    Unsafe unsafe = getUnsafe();

    static Unsafe getUnsafe() {
        try {
            Field f = Unsafe.class.getDeclaredField("theUnsafe");
            f.setAccessible(true);
            return (Unsafe) f.get(null);
        } catch (Exception ignored) {
        }
        try {
            Constructor<Unsafe> c = Unsafe.class.getDeclaredConstructor();
            c.setAccessible(true);
            return c.newInstance();
        } catch (Exception ignored) {
        }
        throw new Error("We need unsafe");
    }
}

File: src/main/java/com/kisman/cc/loader/antidump/UnsafeProvider.kt
package com.kisman.cc.loader.antidump

import sun.misc.Unsafe

/**
 * @author _kisman_
 * @since 12:40 of 04.07.2022
 */
val unsafe: Unsafe by lazy {
    Unsafe::class.java.getDeclaredField("theUnsafe").let {
        it.isAccessible = true
        it[null] as Unsafe
    }
}